src/frontend/utils/utilityTypes.ts

```diff
@@ -0,0 +1,13 @@
+
```

Comment:
I would change the name to partial-record-with-values.type.ts (everywhere we have a dash notation). And please put it to src/utils